### ISSUE TITLE
AuthFailureError on Volley to get beneficiary list

### ISSUE THREAD

**Issue by deepak-ezetap** [cowinapi/developer.cowin#266]
I am trying to get the beneficiary list using Volley on Android Studio.
URL: https://cdn-api.co-vin.in/api/v2/appointment/beneficiaries
Headers {Content-Type=application/json, accept=application/json, Authorization=Bearer <my token>}

The token is received when I confirm the OTP as below:
URL: https://cdn-api.co-vin.in/api/v2/auth/public/confirmOTP
Response JSON: {"token":"eyJhbGciOiJdjflkdjflkd"}

I use the same "token" from confirm OTP response in the header. But error received is com.android.volley.AuthFailureError.

Appreciate your help!

**Comment by deepak-ezetap** [comment]
Also, when I use the "Try it now" option from https://apisetu.gov.in/public/marketplace/api/cowin/cowin-protected-v2#/Vaccination%20Appointment%20APIs/beneficiaries

I get the error Unauthenticated access!

**Comment by xsreality** [comment]
Not sure about the issue but you should redact the JWT token as it contains sensitive information like your mobile number, beneficiary reference id etc.
![image](https://user-images.githubusercontent.com/4991449/118635819-501d9d80-b7d4-11eb-8095-56a4a79ee2e3.png)

**Comment by deepak-ezetap** [comment]
Thanks for letting me know.

**Comment by gautam404** [comment]
> Not sure about the issue but you should redact the JWT token as it contains sensitive information like your mobile number, beneficiary reference id etc.
> ![image](https://user-images.githubusercontent.com/4991449/118635819-501d9d80-b7d4-11eb-8095-56a4a79ee2e3.png)

Can you guide me how you got the Protected API Key

**Comment by xsreality** [comment]
@gautam404 You can decode the token at https://jwt.io

**Comment by vaithitheone** [comment]
@xsreality Did you have access for private APIs, How can i get the api key for private api please share details

**Comment by xsreality** [comment]
@wifirevaithi No, I don't have API keys to access the protected APIs. You will have to contact the Ministry to get it. Not sure what the criteria is to be qualified to receive an API key.

But API key is not the only way to access the protected APIs. Other option is bearer JWT tokens which can be obtained via OTP. Of course not very suitable for automation.

**Comment by deepak-ezetap** [comment]
> @wifirevaithi No, I don't have API keys to access the protected APIs. You will have to contact the Ministry to get it. Not sure what the criteria is to be qualified to receive an API key.
> 
> But API key is not the only way to access the protected APIs. Other option is bearer JWT tokens which can be obtained via OTP. Of course not very suitable for automation.

@xsreality Do we decode the bearer token before adding it in the header? Or can we use it as is?

**Comment by xsreality** [comment]
@deepakezetap you can use the token as is in the Authorization header.

**Comment by deepak-ezetap** [comment]
@xsreality I am getting AuthFailureError 401. I tried the below:
Authorization=token
Authorization=Bearer token

I am trying to get the beneficiaries after confirming the OTP

**Comment by shekharkumar86** [comment]
> @xsreality I am getting AuthFailureError 401. I tried the below:
> Authorization=token
> Authorization=Bearer token
> 
> I am trying to get the beneficiaries after confirming the OTP

i am facing the same issue.if you resolve the beneficiaries list api issue. Please let me know.
